x-pack/filebeat/input/httpjson/config.go

```diff
@@ -17,6 +17,7 @@ type config struct {
 	Request  *requestConfig  `config:"request" validate:"required"`
 	Response *responseConfig `config:"response"`
 	Cursor   cursorConfig    `config:"cursor"`
+	Chain    []chainConfig   `config:"chain"`
```

Comment:
[Question]
`chainConfig` only contains a single field `stepConfig`, is it really needed? Are they both used so the final yaml looks like that:
```yaml
   chain:
     - step:
         request.url: https://some_url.com/services/data/v1.0/$.records[:].id/export_ids (second call)
         request.method: GET
         replace: $.records[:].id
     - step:
         request.url: https://some_url.com/services/data/v1.0/export_ids/$.file_name/info (third call)
         request.method: GET
         replace: $.file_name
```

Maybe we could use the `stepConfig` directly:
```suggestion
	Chain    []stepConfig   `config:"chain"`
```
I believe it would result in a yaml like this:
```yaml
chain:
  - request.url: 'https://some_url.com/services/data/v1.0/$.records[:].id/export_ids'
    request.method: GET
    replace: '$.records[:].id'
  - request.url: 'https://some_url.com/services/data/v1.0/export_ids/$.file_name/info'
    request.method: GET
    replace: $.file_name

```

Reply:
Yes, this is related to https://github.com/elastic/beats/pull/29816#discussion_r815601148. Changing type would be better than just documenting the situation, though does this preclude extending the computation model in the future, and if it does, does that matter.

Reply:
Right, we thought of that but then decided to go with using a step to make it easy in the future to extend it to add some more processing in-between steps whenever required, for example:
```- type: httpjson
  enabled: true
  request.url: https://example.com/
  interval: 1h
  chain:
	- step:
		request.method: GET
		request.url: "http://example.com/auth"
		response: ...

	- break:
		when: '[[eq .last_response.body.authtok ""]]'

	- step:
		request.method: POST
		request.url: 'http://example.com/api?auth=[[.last_response.body.authtok]]'
		response: ...```
Hence, we were planning to update the docs for the user to mention using steps.